File: src/main/java/org/pl/lexer/LexerError.java
package org.pl.lexer;

public class LexerError {

    private final String message;
    private final int location;

    public LexerError(String message, int location) {
        this.message = message;
        this.location = location;
    }

    @Override
    public String toString() {
        return new StringBuilder("Lexer error at position ")
                .append(location)
                .append(": ")
                .append(message)
                .toString();
    }
}

File: src/main/java/org/nyxlang/lexer/LexerError.kt
package org.nyxlang.lexer


/**
 * A lexer error indicates at which [location] in the program code some error occurred. This
 * error is described by some given [message].
 */
class LexerError(private val message: String?, private val location: Int) {
    override fun toString() = "Lexer error at position $location: $message"
}